### PR TITLE
add stereoset

### PR DESCRIPTION
**Pull request by cstorm125**
StereoSet is a dataset that measures stereotype bias in language models. StereoSet consists of 17,000 sentences that measures model preferences across gender, race, religion, and profession.

### CODE REVIEW

File: datasets/stereoset/stereoset.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                        {
+                            "sentence": datasets.Value("string"),
+                            "id": datasets.Value("string"),
+                            "labels": datasets.features.Sequence(
+                                {
+                                    "label": datasets.Value("string"),
+                                    "human_id": datasets.Value("string"),
+                                }
+                            ),
+                            "gold_label": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/stereoset/stereoset.py:87]: Can you use the ClassLabel feature type here ?

For example
```
"gold_label": datasets.ClassLabel(names=["anti-stereotype", "stereotype", "unrelated", "related"])
```

you'll need to regenerate the dataset_infos.json file after this change

File: datasets/stereoset/stereoset.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    def __init__(self, **kwargs):
+        """BuilderConfig
+
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(StereoSetConfig, self).__init__(**kwargs)
+
+
+class StereoSet(datasets.GeneratorBasedBuilder):
```
Comment (lhoestq) [inline datasets/stereoset/stereoset.py:53]:
```suggestion
class Stereoset(datasets.GeneratorBasedBuilder):
```
this must be the camel case version of the dataset script name `stereoset`

File: datasets/stereoset/README.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+- `target`: target for stereotype
+- `bias_type`: type of biases including gender, race, religion, and profession
+- `context`: context sentence
+- `sentences`
+  - `sentence`: choice of sentence for given context and target
+  - `id`: sentence id
+  - `labels`:
+    - `label`: label of sentence including stereotype, anti-stereotype, and unrelated
+    - `human_id`: id of annotator
+  - `gold_label`: gold label of the question
```
Comment (lhoestq) [inline datasets/stereoset/README.md:93]: can you describe briefly which values this field can take and their meanings ?

File: datasets/stereoset/README.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+multilinguality:
+- monolingual
+size_categories:
+- 1K<n<10K
+source_datasets:
+- original
+task_categories:
+- question-answering
+task_ids:
+- multiple-choice-qa
```
Comment (lhoestq) [inline datasets/stereoset/README.md:19]: This doesn't look like QA to me
What about

```suggestion
task_categories:
- text-classification
task_ids:
- text-classification-other-stereotype-detection
```

**Review by lhoestq — COMMENTED** [review]
Thanks !

I left a few comments

File: datasets/stereoset/stereoset.py (hunk 1 above)
Comment (cstorm125) [reply datasets/stereoset/stereoset.py:87]: I replaced `datasets.Value("string")` with `datasets.ClassLabel(names=["anti-stereotype", "stereotype", "unrelated"])` for both `label` of individual humans and `gold_label`. (there is no "related" value for both columns)

**Review by lhoestq — APPROVED** [review]
Thank you !

**Pull request merged**